File: Zisimopoulos_Valencia/Software Construction/QLS/src/uva/sc/qls/logic/style/FontSize.java
package uva.sc.qls.logic.style;

import uva.sc.qls.ast.IQLSNodeVisitor;

public class FontSize implements StyleProperty {

	Integer	value;

	public FontSize(Integer value) {
		this.value = value;
	}

	public <T> T accept(IQLSNodeVisitor<T> visitor) {
		return visitor.visit(this);
	}

	@Override
	public Integer getValue() {
		return value;
	}

	public String toString() {
		String result = "[FontSize]: " + value;
		return result;
	}

}

File: CarlaErik/UvA.SoftCon.Questionnaire/QLS/AST/Model/StyleAttributes/FontSize.cs
using System;
using UvA.SoftCon.Questionnaire.Common.AST.Model;
using UvA.SoftCon.Questionnaire.QLS.StyleSets;

namespace UvA.SoftCon.Questionnaire.QLS.AST.Model.StyleAttributes
{
    public class FontSize : StyleAttribute
    {
        public int Size
        {
            get;
            private set;
        }

        internal FontSize(int size, TextPosition position)
            :base(position)
        {
            Size = size;
        }

        public override T Accept<T>(IStyleSheetVisitor<T> visitor)
        {
            return visitor.VisitFontSize(this);
        }

        public override void OverrideStyle(StyleSet styleSet)
        {
            styleSet.OverrideFontSize(Size);
        }

        public override string ToString()
        {
            return String.Format("{0}pt", Size);
        }
    }
}
